themes/default/content/blog/advanced-aws-networking-part-2/index.md

```diff
@@ -0,0 +1,534 @@
+
+<!--more-->
+
+## Introduction
+
+In [AWS Advanced Networking Part 1](https://pulumi.com/blog/advanced-aws-networking-part-1/):
+
+1. Discussed the benefits of a hub-and-spoke network architecture with centralized egress.
+1. Created the shared [AWS Transit Gateway](https://aws.amazon.com/transit-gateway/) resources to be shared among our spoke VPCs and inspection VPC.
+1. Defined our inspection VPC as a Pulumi [component resource](https://www.pulumi.com/docs/intro/concepts/resources/components/).
+
+Now, to finish the architecture, we will need to:
+
+1. Create a spoke VPC as a Pulumi component resource.
+1. Test the network architecture by verifying that centralized egress is working.
+1. Add traffic inspection with [AWS Network Firewall](https://aws.amazon.com/network-firewall/).
+
+{{% notes type="info" %}}
+In subsequent sections we will be showing code samples. Because this is a more advanced Pulumi program, we will be focusing on the most important parts of the architecture rather than covering every single line in the codebase. The complete code for this project is available on [GitHub](https://github.com/pulumi/examples/tree/master/aws-py-hub-and-spoke-network), which is a great place to go for full context.
+{{% /notes %}}
+
+## Spoke VPC
+
+Like your inspection VPC, we will implement the spoke VPC as a Pulumi component resource. Component resources allow you to logically group resources to form reusable components. Because your network will contain multiple spoke VPCs (at least 2, possibly more), and the spoke VPC will contain both the basic VPC as well as connectivity to Transit Gateway, the spoke VPC is a natural fit for a component resource.
+
+First, add a file `spoke.py` to your Pulumi program. In Python, it's good practice to put each component resource along with its associated arguments class in a separate file from your main program as it makes for intuitive code organization:
+
+```python
+@dataclass
+class SpokeVpcArgs:
+    vpc_cidr_block: str
+    tgw_id: pulumi.Input[str]
+    tgw_route_table_id: pulumi.Input[str]
+
+
+class SpokeVpc(pulumi.ComponentResource):
+    def __init__(self, name: str, args: SpokeVpcArgs, opts: pulumi.ResourceOptions = None) -> None:
+        super().__init__("awsAdvancedNetworking:index:SpokeVpc", name, None, opts)
+```
+
+Then, create a VPC with 2 sets of subnets, both isolated (that is, with no pathway to the internet). One set of subnets will hold our spoke VPC workloads, and the other will hold our Transit Gateway attachment. We enable DNS hostnames and support so that we can later use VPC endpoints to enable [AWS Systems Manager](https://docs.aws.amazon.com/systems-manager/index.html) access to verify that centralized egress is working as expected without requiring direct host access via SSH:
```

Comment:
```suggestion
Then, create a VPC with two sets of subnets, both isolated (with no pathway to the internet). One set of subnets will hold our spoke VPC workloads, and the other will hold our Transit Gateway attachment. We enable DNS hostnames and support so that we can later use VPC endpoints to enable [AWS Systems Manager](https://docs.aws.amazon.com/systems-manager/index.html) access to verify that centralized egress is working as expected without requiring direct host access via SSH:
```